### PR TITLE
[4.0][api][com_plugins] - Add filter options

### PR DESCRIPTION
**Pull request by alikon**
### Summary of Changes
added the option to filter plugins by element, state, search,folder

### Testing Instructions
call `JROOT_URL/api/index.php/v1/plugins?filter[element]=privacyconsent`
call `JROOT_URL/api/index.php/v1/plugins?filter[type]=system`
call `JROOT_URL/api/index.php/v1/plugins?filter[search]=Web Services`
call `JROOT_URL/api/index.php/v1/plugins?filter[status]=0`
or any combination like : 
`JROOT_URL/api/index.php/v1/plugins?filter[type]=system&filter[status]=0`
etc

### Expected result
the api response is filtered 

### Actual result
N/A

### Documentation Changes Required
Yes

### CODE REVIEW

File: api/components/com_plugins/src/Controller/PluginsController.php [hunk 1]
```diff
@@ -90,4 +91,39 @@ public function edit()
...
+		{
+			$this->modelState->set('filter.enabled', $filter->clean($apiFilterInfo['enabled'], 'INT'));
+		}
+
+		if (array_key_exists('search', $apiFilterInfo))
+		{
+			$this->modelState->set('filter.search', $filter->clean($apiFilterInfo['search'], 'STRING'));
+		}
+
+		if (array_key_exists('folder', $apiFilterInfo))
```
Comment (wilsonge) [inline api/components/com_plugins/src/Controller/PluginsController.php]: Let's call this type please in the API (as it matches what we call it in the UI)

Comment (alikon) [reply api/components/com_plugins/src/Controller/PluginsController.php]: yes make sense
done

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on ef49b89e53ab0b3aaf0b1ba218994d82d37cf2c6<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/28418">issues.joomla.org/tracker/joomla-cms/28418</a>.</sub>

**Comment by wilsonge** [comment]
Thanks!

**Comment by Razzo1987** [comment]
I have tested this item :white_check_mark: successfully on ef49b89e53ab0b3aaf0b1ba218994d82d37cf2c6<br /><br />All listed GET successfully tested<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/28418">issues.joomla.org/tracker/joomla-cms/28418</a>.</sub>

**Pull request merged**